src/uuid/rng/init.lua

```diff
@@ -78,7 +78,42 @@ function rng.luasystem()
   return sys.random
 end
 
+----------------------------------------------------------------------------
+-- Returns an rng that implements Windows' RtlGenRandom function which is
+-- a good source of randomness on all modern versions of Windows.
+-- Requires LuaJIT or a LuaJIT-compatible FFI library.
+-- @treturn function A function that returns `n` random bytes, signature: `byte_string, err = func(n)`
+-- @usage
+-- local uuid = require "uuid"
+-- uuid.set_rng(uuid.rng.win_ffi())
+function rng.win_ffi()
+  local ok, ffi = pcall(require, "ffi")
+  if not ok then 
+      ok, ffi = pcall(require, "cffi")
+  end
+  if not ok then
+    return nil, "ffi not available"
+  end
 
+  local ffi_defs = [[
+    unsigned char SystemFunction036(void *RandomBuffer, unsigned long RandomBufferLength);
+  ]]
+  ffi.cdef(ffi_defs)
```

Comment:
this will probably fail if you call it again

Reply:
This testing code shows it simply appends to the definitions:
```lua
do
    local ffi = require("ffi")
    ffi.cdef[[
        void rand();
    ]]
end
do
    local ffi = require("ffi")
    ffi.cdef[[
        void memcpy();
    ]]
end
local ffi = require("ffi")
print(ffi.C.rand)
print(ffi.C.memcpy)
```

Reply:
although it's not great to keep re-defining the same symbol over and over again, maybe some caching could be done.

Reply:
yes caching such that we define it only once would be great